test/e2e/backup_restore_test.go

```diff
@@ -0,0 +1,231 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/iancoleman/strcase"
+	v1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	v2 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v2alpha1"
+	cconsts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	ispn "github.com/infinispan/infinispan-operator/pkg/infinispan"
+	ispnclient "github.com/infinispan/infinispan-operator/pkg/infinispan/client/http"
+	tconst "github.com/infinispan/infinispan-operator/test/e2e/constants"
+	tutils "github.com/infinispan/infinispan-operator/test/e2e/utils"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type clusterSpec func(name, namespace string, clusterSize int) *v1.Infinispan
+
+func TestBackupRestore(t *testing.T) {
+	t.Run(string(v1.ServiceTypeDataGrid), testBackupRestore(datagridService))
+	t.Run(string(v1.ServiceTypeCache), testBackupRestore(cacheService))
+}
+
+func testBackupRestore(clusterSpec clusterSpec) func(*testing.T) {
+	return func(t *testing.T) {
+		// Create a resource without passing any config
```

Comment:
parallel should be enabled, it cause a problem
```suggestion
		t.Parallel()
		// Create a resource without passing any config
```